libs/s25main/WindowManager.cpp

```diff
@@ -871,3 +871,51 @@ void WindowManager::DrawToolTip()
         curTooltip->draw(ttPos);
     }
 }
+
+DrawPoint WindowManager::SnapWindow(unsigned wndId, const Rect& wndRect) const
+{
+    constexpr auto snapDistance = 15;
+    DrawPoint minDist(snapDistance + 1, snapDistance + 1);
+    DrawPoint minSnapOffset = DrawPoint::all(0);
+    auto calcSnapOffset = [](int a, int b, int& minDist, int& snapOffset) {
+        int diff = b - a;
+        int dist = std::abs(diff);
+        if(dist < minDist)
+        {
+            minDist = dist;
+            snapOffset = diff;
+        }
+    };
+
+    for(const auto& wnd : windows)
+    {
+        if(wnd->GetID() == wndId)
+            continue;
+
+        DrawPoint dist(snapDistance + 1, snapDistance + 1);
+        DrawPoint snapOffset = DrawPoint::all(0);
+        const auto calcSnapOffsetX = [&](int a, int b) { calcSnapOffset(a, b, dist.x, snapOffset.x); };
+        const auto calcSnapOffsetY = [&](int a, int b) { calcSnapOffset(a, b, dist.y, snapOffset.y); };
+        const Rect rect = wnd->GetDrawRect();
```

Comment:
Should I use `GetBoundaryRect()` instead? What's the practical difference between the two?

Reply:
For windows it doesn't matter: They are the same. Only controls may have a different size, e.g. size of text-controls depend on the current text.

For correctness here I'd say `GetBoundaryRect` looks more sensible: We want to actual, current extents of the window (in case we implement dynamic windows later)